doc/explanation/opam-integration.rst

```diff
@@ -0,0 +1,115 @@
+How Dune integrates with opam
+=============================
+
+When instructed to do so (see :doc:`../howto/opam-file-generation`), Dune generates opam files with the following instructions:
+
+.. code::
+
+   build: [
+     ["dune" "subst"] {dev}
+     [
+       "dune"
+       "build"
+       "-p"
+       name
+       "-j"
+       jobs
+       "@install"
+       "@runtest" {with-test}
+       "@doc" {with-doc}
+     ]
+   ]
+
+Let's see what this means in detail.
+
+Substitution
+------------
+
+The first step is to call ``dune subst``, but only if the ``{dev}`` opam
+variable is set. This variable is only set when the package is pinned.
+This means that :ref:`dune-subst` does not run for released versions, but it
+does for development versions.
+
+This is not a problem since released versions should have a ``(version)`` field set in ``dune-project``, and placeholder substitution should have been performed. `dune-release`_ takes care of these steps.
+
+.. _dune-release: https://github.com/tarides/dune-release
+
+Opam Variables
+--------------
+
+In the second command line, ``name`` is a variable that evaluates to the name
+of the package being built and ``jobs`` is a variable that corresponds to the
+number of commands to run in parallel.
+
+What ``-p`` Means
+-----------------
+
+The ``-p`` flag, shorthand for ``--release-of-packages``, is Dune's public interface to set up the options for an opam build. The exact semantics may change, but as of Dune 3.8 it is equivalent to the combination of:
+
+- ``--root .``: set the :term:`root` to prevent Dune from :ref:`looking it up <finding-root>`.
+- ``--only-packages name``: ignore packages other than ``name`` defined in the project.
+- ``--profile release``: set the :term:`build profile` to ``release``. In particular, this ensures that warnings are not fatal.
+- ``--ignore-promoted-rules``: silently ignores all rules with ``(mode promote)``.
+- ``--default-target @install``: make sure that ``dune build`` with no target argument builds ``@install``, not ``@@default`` (this is not used in the opam integration since an explicit target is passed)
+- ``--no-config``: do not load the configuration file in the user's home directory.
```

Comment:
it's poorly worded but I was going for an imperative here, as in:

> we're passing options X, Y and Z:
> - X: do this thing
> - Y: make sure that something happens
> - Z: inform that component that we're doing this

what would be a good style to convey that info?

Reply:
I see! I can see what you mean by going for the imperative. If your reader will understand that (i.e., other engineers), you can leave it as it is. The way it reads to me is “name of the thing…this thing does…”

It would be too repetitive to say “Use xyz to do this thing” in each one…
I’m not quite sure… 

Any thoughts on your end?

Reply:
I think I'll keep it worded like that for now. We could use a table I suppose but that would be even more engineer-y :/